src/scripts/main.js

```diff
@@ -1,3 +1,266 @@
 'use strict';
 
-// write your code here
+const gameContainer = document.querySelector('.container');
+const startButton = gameContainer.querySelector('.button');
+const cells = gameContainer.querySelectorAll('.field-cell');
+const messageStart = gameContainer.querySelector('.message-start');
+const messageWin = gameContainer.querySelector('.message-win');
+const messageLose = gameContainer.querySelector('.message-lose');
+const gameScore = gameContainer.querySelector('.game-score');
+const gameRow = gameContainer.querySelectorAll('tr');
+const probabilityGeneratingNumber = 0.1;
+const numberFour = 4;
+const numberTwo = 2;
+const moveLeft = 'ArrowLeft';
+const moveRight = 'ArrowRight';
+const moveUp = 'ArrowUp';
+const moveDown = 'ArrowDown';
+const winnerNumber = '2048';
+let movePossible = false;
+let adjacentCells = false;
+
+function generateRandom() {
+  const randomI = Math.floor(Math.random() * cells.length);
+
+  if (cells[randomI].textContent === '') {
+    const randomNum = Math.random() < probabilityGeneratingNumber
+      ? numberFour : numberTwo;
+
+    cells[randomI].textContent = randomNum;
+    cells[randomI].classList.add('field-cell--' + randomNum);
+  } else {
+    generateRandom();
+  }
+}
+
+startButton.addEventListener('click', () => {
+  if (startButton.classList.contains('start')) {
+    generateRandom();
+    generateRandom();
+    messageStart.classList.add('hidden');
+    startButton.classList.remove('start');
+    startButton.classList.add('restart');
+    startButton.textContent = 'Restart';
+  } else if (startButton.classList.contains('restart')) {
+    cells.forEach((cell) => {
+      cell.textContent = '';
+      cell.className = 'field-cell';
+    });
+    messageStart.classList.remove('hidden');
+    startButton.classList.remove('restart');
+    startButton.classList.add('start');
+    startButton.textContent = 'Start';
+    gameScore.textContent = '0';
+    messageLose.classList.add('hidden');
+    messageWin.classList.add('hidden');
+  }
+});
+
+function updateCells() {
+  cells.forEach((cell) => {
+    if (cell.textContent === '') {
+      cell.className = 'field-cell';
+    } else {
+      cell.className = `field-cell field-cell--${cell.textContent}`;
+    }
+  });
+}
+
+document.addEventListener('keydown', (e) => {
+  switch (e.key) {
+    case moveLeft:
+      slideLeft();
+      break;
+
+    case moveRight:
+      slideRight();
+      break;
+
+    case moveUp:
+      slideUp();
+      break;
+
+    case moveDown:
+      slideDown();
+      break;
+  }
+
+  checkGameWin();
+  checkGameOver();
+});
+
+function filterZero(cellsArray) {
+  return cellsArray.filter(i => i !== '');
+}
+
+function slide(cellsArray) {
+  let arrayToSlide = filterZero(cellsArray);
+  let updatedScore = 0;
+
+  for (let i = 0; i < arrayToSlide.length; i++) {
+    if (arrayToSlide[i] === arrayToSlide[i + 1]) {
+      arrayToSlide[i] *= 2;
```

Comment:
You are reassigning an array value directly. This can lead to mutation of the original array. It's a better practice to create a new array with the modified elements.